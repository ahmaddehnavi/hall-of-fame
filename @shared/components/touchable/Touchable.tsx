import * as React from 'react';
import {Platform, TouchableNativeFeedback, TouchableOpacity, TouchableWithoutFeedback, TouchableWithoutFeedbackProps, View} from 'react-native';


export type TouchableProps = TouchableWithoutFeedbackProps & {
    circle?: boolean
    withoutFeedback?: boolean
    children: React.ReactNode
}

export function Touchable(props: TouchableProps) {

    if (props.withoutFeedback) {
        return (
            <TouchableWithoutFeedback
                {...props}
            />
        );
    }
    if (Platform.OS === 'android') {
        if (Platform.Version >= 21) {
            return (
                <TouchableNativeFeedback
                    {...props}
                    background={
                        props.circle ?
                            TouchableNativeFeedback.SelectableBackgroundBorderless() :
                            TouchableNativeFeedback.SelectableBackground()
                    }
                />
            )
        }
        return (
            <TouchableNativeFeedback
                {...props}
                background={undefined}
            />
        )
    }

    return (
        <TouchableOpacity {...props}/>
    )
}