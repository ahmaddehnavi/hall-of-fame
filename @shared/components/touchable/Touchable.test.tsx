import React from 'react';
import 'react-native';
import {View} from 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import {Touchable} from './Touchable';

describe('Touchable Tests', () => {

    it('should match snapshot', () => {
        let tree = renderer.create(
            <Touchable>
                <View/>
            </Touchable>
        );
        expect(tree).toMatchSnapshot();
    });

});
