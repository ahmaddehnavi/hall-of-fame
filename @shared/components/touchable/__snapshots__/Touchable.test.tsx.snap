// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Touchable Tests should match snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View />
</View>
`;
